cli/flox-rust-sdk/src/models/lockfile.rs

```diff
@@ -490,44 +503,41 @@ impl LockedManifestCatalog {
         manifest: &'manifest TypedManifestCatalog,
         groups: impl IntoIterator<Item = ResolvedPackageGroup> + 'manifest,
     ) -> Result<impl Iterator<Item = LockedPackageCatalog> + 'manifest, LockedManifestError> {
-        // For each group, extract the first page and its system.
-        // Error if the first page doesn't contain any packages.
-        let first_pages: Vec<(Vec<PackageResolutionInfo>, String)> = groups
+        Ok(groups
             .into_iter()
-            .map(|mut group| {
-                group
-                    .pages
-                    .first_mut()
-                    .and_then(|page| {
-                        std::mem::take(&mut page.packages)
-                            .map(|packages| (packages, group.system.clone()))
-                    })
-                    .ok_or_else(|| {
-                        LockedManifestError::NoPackagesOnFirstPage(group.name, group.system)
-                    })
+            .map(|group| {
+                if let Some(page) = group.page {
+                    if !page.complete {
+                        return Err(LockedManifestError::ResolutionFailed(
+                            "page wasn't complete".into(),
+                        ));
+                    }
+                    if let Some(pkgs) = page.packages {
+                        Ok(pkgs)
+                    } else {
+                        Err(LockedManifestError::EmptyPage)
+                    }
+                } else {
+                    Err(LockedManifestError::ResolutionFailed(
+                        "package group had no page".into(),
+                    ))
```

Comment:
suggestion (nb): identify group name, include message from catalog service
suggestion: remove mention of pages (unknown/internal concept)